src/Nethermind/Nethermind.Serialization.Rlp/BlockDecoder.cs

```diff
@@ -9,9 +9,16 @@ namespace Nethermind.Serialization.Rlp
 {
     public class BlockDecoder : IRlpValueDecoder<Block>, IRlpStreamDecoder<Block>
     {
-        private readonly HeaderDecoder _headerDecoder = new();
+        private readonly IHeaderDecoder _headerDecoder;
         private readonly BlockBodyDecoder _blockBodyDecoder = BlockBodyDecoder.Instance;
 
+        public BlockDecoder() : this(new HeaderDecoder()) { }
+
+        public BlockDecoder(IHeaderDecoder headerCodec)
```

Comment:
`headerCodec` -> `headerDecoder`

Also I would make it primary constructor

Reply:
Sorry about the naming, forgot it was there in the previous change 😅